core/sts-3.3.md

```diff
@@ -0,0 +1,189 @@
+---
+title: IRCv3.3 Strict Transport Security
+layout: spec
+copyrights:
+  -
+    name: "Attila Molnar"
+    period: "2015"
+    email: "attilamolnar@hush.com"
+---
+
+## Description
+
+Strict Transport Security (STS) is a mechanism which allows servers to
+communicate that complying clients should only connect to them over a secure
+connection.
+
+The policy is communicated to the client via the `sts` capability and should
+be processed by the client at capability negotiation time.
+
+The value of the capability specifies the duration during which the client
+MUST only connect to the server securely (using TLS, aka. SSL), and the port
+number to use for it.
+
+Clients MUST use sufficient certificate verification to ensure the connection
+is secure without any errors.
+
+## Details
+
+The capability value, which MUST always be present, is a comma (`,`) separated
+list of tokens. Each token is a key which may have a value attached. If there
+is a value attached, the value is separated from the key by a `=`. That is,
+`<key>[=<value>][,<key2>[=<value2>][,<keyN>[=<valueN>]]]`. Keys specified in
+this document MUST only occur at most once.
+
+Clients MUST ignore every token with a key that they don't understand.
+
+An `sts` capability having at least a `duration` key expresses an STS policy.
+
+See [capability negotiation 3.2](capability-negotiation-3.2.html) for more
+information about capabilities with values.
+
+### Mechanism
+
+When the client sees the `sts` capability advertised over a non-secure
+connection, it MUST first establish a secure connection and confirm that the
+`sts` capability is still present.
+
+Once the client has confirmed that the `sts` capability is indeed offered over
+a secure connection, it then MUST only attempt secure connections to the
+server from now on until the policy expires (see the `duration` key).
+It MUST refuse to connect if a secure connection cannot be established with the
+server for any reason during the lifetime of the policy.
+
+However, if the client fails to connect securely and get the STS policy over a
+secure connection for any reason it SHOULD retry the secure connection next
+time it receives the `sts` cap over a non-secure connection.
+
+Clients MUST NOT request this capability (`CAP REQ`).
+Servers MUST reply with an appropriate `CAP NAK` message if a client requests
+this capability.
+
+### The `duration` key
+
+This key indicates how long from receiving the advertisement clients MUST use a
+secure connection when connecting to this server. This MUST be specified in
+seconds as the value of this key and as a single integer without a prefix or
+suffix.
+
+Servers MUST send this key to securely connected clients.
+Servers SHOULD NOT send this key to non-securely connected clients, because it
+will be ignored.
```

Comment:
For server it's easier to just use the same string in CAP LS, without caring whether client connected with ssl or not.